fv3core/stencils/fxadv.py

```diff
@@ -1,938 +1,677 @@
-        grid().ie,
-        grid().je,
-        grid().ie,
-        grid().je,
-        west=False,
-        lower=True,
-        south=False,
-        vswitch=True,
-    )
-
-
-def nw_corner(uc, vc, ut, vt, cosa_u, cosa_v, corner_shape):
-    t = grid().js + 1
-    n = grid().js
-    z = grid().js - 1
-    corner_ut(
-        uc,
-        vc,
-        ut,
-        vt,
-        cosa_u,
-        cosa_v,
-        t,
-        grid().je + 1,
-        n,
-        grid().je + 2,
-        west=True,
-        lower=False,
-    )
-    corner_ut(
-        vc,
-        uc,
-        vt,
-        ut,
-        cosa_v,
-        cosa_u,
-        z,
-        grid().je,
-        z,
-        grid().je,
-        west=True,
-        lower=False,
-        south=False,
-        vswitch=True,
-    )
-    corner_ut(
-        uc,
-        vc,
-        ut,
-        vt,
-        cosa_u,
-        cosa_v,
-        t,
-        grid().je,
-        n,
-        grid().je,
-        west=True,
-        lower=True,
-    )
-    corner_ut(
-        vc,
-        uc,
-        vt,
-        ut,
-        cosa_v,
-        cosa_u,
-        n,
-        grid().je,
-        t,
-        grid().je,
-        west=True,
-        lower=True,
-        south=False,
-        vswitch=True,
-    )
-
-
-# TODO: Probably can delete -- but in case we want to do analysis to show it
-# doesn't matter at all.
-# {
+@gtstencil()
+def fxadv_stencil(
+    cosa_u: FloatField,
+    cosa_v: FloatField,
+    rsin_u: FloatField,
+    rsin_v: FloatField,
+    sin_sg1: FloatField,
+    sin_sg2: FloatField,
+    sin_sg3: FloatField,
+    sin_sg4: FloatField,
+    rdxa: FloatField,
+    rdya: FloatField,
+    area: FloatField,
+    dy: FloatField,
+    dx: FloatField,
+    uc: FloatField,
+    vc: FloatField,
+    crx_adv: FloatField,
+    cry_adv: FloatField,
+    xfx_adv: FloatField,
+    yfx_adv: FloatField,
+    ut: FloatField,
+    vt: FloatField,
+    ra_x: FloatField,
+    ra_y: FloatField,
+    dt: float,
+):
+    """
+    Updates flux operators and courant numbers for fvtp2d
+    Inputs:
+        uc: x-velocity on the C-grid
+        vc: y-velocity on the C-grid
+        crx_adv: Courant number, x direction(inout)
+        cry_adv: Courant number, y direction(inout)
+        xfx_adv: Finite volume flux form operator in x direction (inout)
+        yfx_adv: Finite volume flux form operator in y direction (inout)
+        ut: temporary x-velocity transformed from C-grid to D-grid equivalent(?) (inout)
+        vt: temporary y-velocity transformed from C-grid to D-grid equivalent(?) (inout)
+        ra_x: Area increased in the x direction due to flux divergence (inout)
+        ra_y: Area increased in the y direction due to flux divergence (inout)
+        dt: timestep in seconds
+    Grid variable inputs:
+        cosa_u, cosa_v, rsin_u, rsin_v, sin_sg1,sin_sg2, sin_sg3, sin_sg4,
+        rdxa, rdya, area, dy, dx
+    Returns:
+       Updates to xfx_adv, yfx_adv, crx_adv, cry_adv, ut, vt, ra_x, ra_y
```

Comment:
I like this descriptive docstring! I really like that the grid variables and returns are documented. One nit and then I think we can merge this. It doesn't follow the docstring format https://github.com/VulcanClimateModeling/fv3core/blob/master/CONTRIBUTING.md#docstrings. Napoleon rules dictate that the only sections allowed are those listed here: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/#docstring-sections. Can we reformat this following those rules?

Reply:
Ok sure, will do, I just imitated other docstrings I saw in the repo